src/Components/Details/Details.js

```diff
@@ -1,62 +1,58 @@
-import React from 'react'
-import './Details.css'
+import React, { Component } from 'react'
+import { fetchMovieDetails, fetchMovieVideos } from '../../apiCalls'
+import DetailsDisplay from '../DetailsDisplay/DetailsDisplay'
 import backArrow from './double-left-arrows.svg'
+import './Details.css'
 
-const Details = ({ allMovies, selectedMovieID, hideSelection }) => {
-
-  const movieToDisplay = allMovies.find(movie => {
-    return movie.id === selectedMovieID
-  })
-
-  console.log('MOVIE >>> ', movieToDisplay)
-
-  const releaseYear = movieToDisplay.release_date.split('-')[0]
-  const rating = movieToDisplay.average_rating.toFixed(1)
-  const genres = movieToDisplay.genres.map(genre => {
-  return <p className='details__each-genre'>{genre}</p>
-  })
-
-  const clearID = () => {
-    hideSelection(null)
+class Details extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedMovieID: props.selectedMovieID,
+      movieToDisplay: {},
+      isLoading: true,
+      error: "",
+      trailerToDisplay: {}
+    }
   }
 
-  return (
-    // <>
-    <section className='details__details-view'>
-      <button className='return-to-home-view-btn' onClick={clearID}>
+  componentDidMount() {
+    fetchMovieDetails(this.state.selectedMovieID)
+      .then(movie => {
+        this.setState({ movieToDisplay: movie.movie, isLoading: false })
+        this.findVideos()
+      })
+      .catch(error => this.setState({ error: "These taters got too rowdy - check back later!"}))
+ 
+  }
+  findVideos = () => {
+    fetchMovieVideos(this.state.selectedMovieID)
+      .then(trailers => this.setState({ trailerToDisplay: trailers.videos }))
+      .catch(error =>  this.setState({error: "These taters got too rowdy - check back later!"}) )
+  }  
+
+  clearID = () => {
+    this.props.hideSelection(null)
+  }
+  
+  render() {
+    return (
+      <section className='details__details-view'>
+        <button className='return-to-home-view-btn' onClick={this.clearID}>
```

Comment:
Really smart idea to break our Details out to a new component to render like this, and definitely feels like it cleans that up a bit in this file!